packages/zoltan2/core/src/algorithms/color/Zoltan2_AlgHybrid2GL.hpp

```diff
@@ -0,0 +1,1532 @@
+    using base_adapter_t = typename Adapter::base_adapter_t;
+    using map_t = Tpetra::Map<lno_t,gno_t>;
+    using femv_scalar_t = int;
+    using femv_t = Tpetra::FEMultiVector<femv_scalar_t, lno_t, gno_t>; 
+    using device_type = Tpetra::Map<>::device_type;
+    using execution_space = Tpetra::Map<>::execution_space;
+    using memory_space = Tpetra::Map<>::memory_space;
+    double timer(){
+      struct timeval tp;
+      gettimeofday(&tp, NULL);
+      return ((double) (tp.tv_sec) + 1e-6 * tp.tv_usec);
+    }
+  private:
+    
+    void buildModel(modelFlag_t &flags);
+    
+    //Entry point for parallel local coloring
+    //  nVtx is the number of vertices owned by the current process
+    //
+    //  adjs_view is the adjacencies, indexed by offset_view
+    //
+    //  offset_view is the CSR row map, used to index the adjs_view
+    //
+    //  femv is the FEMultiVector that holds the colors for the vertices
+    //       the colors will change in this function.
+    //
+    //  vertex_list is a list of vertices to recolor
+    //
+    //  vertex_list_size is the size of the list of vertices to recolor
+    //                   vertex_list_size = 0 means recolor all uncolored vertices
+    //
+    //  recolor decides which KokkosKernels algorithm to use.
+    //
+    virtual void colorInterior(const size_t nVtx,
+                       Kokkos::View<lno_t*, device_type > adjs_view,
+                       Kokkos::View<offset_t*,device_type > offset_view,
+                       Teuchos::RCP<femv_t> femv,
+		       Kokkos::View<lno_t*, device_type> vertex_list,
+		       size_t vertex_list_size = 0,
+                       bool recolor=false) = 0;
+    
+    //Entry point for serial local coloring
+    virtual void colorInterior_serial(const size_t nVtx,
+                       typename Kokkos::View<lno_t*, device_type >::HostMirror adjs_view,
+                       typename Kokkos::View<offset_t*,device_type >::HostMirror offset_view,
+                       Teuchos::RCP<femv_t> femv,
+		       typename Kokkos::View<lno_t*, device_type>::HostMirror vertex_list,
+		       size_t vertex_list_size = 0,
+                       bool recolor=false) = 0;
+  public:
+    //Entry point for parallel conflict detection
+    //  n_local is the number of vertices owned by the current process
+    //
+    //  dist_offsets_dev is the device view that holds the CSR offsets
+    //
+    //  dist_adjs_dev is the device view that holds CSR adjacencies
+    //
+    //  femv_colors is the device color view.
+    //              colors are modified in this function.
+    //
+    //  boundary_verts_view holds the local IDs of vertices in the boundary
+    //
+    //  boundary_size is the number of entries in boundary_verts_view
+    //
+    //  verts_to_recolor_atomic is an atomic device view that holds the list
+    //                          of vertices to recolor
+    //
+    //  verts_to_recolor_size_atomic is an atomic device view that holds the
```

Comment:
Does verts_to_recolor_size_atomic hold the allocated size of verts_to_recolor_atomic, or the number of vertices to be recolored?  Also would be good to say why this variable is a view rather than a size_t.